apps/backend/app/api/routes/user_data.py

```
@@ -92,27 +91,16 @@ async def get_preview_data(user_id: str = Depends(get_current_user_id)) -> Dict[
-        print(f"Original S3 URL: {s3_url}")
-
-        # Handle different S3 URL formats
-        # Parse the S3 URL to extract the full object key (preserving directory prefixes)
-        parsed_url = urlparse(s3_url)
-        s3_key = parsed_url.path.lstrip("/")
-
-        if not s3_key:
+        try:
+            _, s3_key = parse_s3_url(s3_url)
+        except ValueError:
             raise HTTPException(status_code=400, detail="Invalid S3 URL (missing object key)")
```

Ruff (0.15.15)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)